From 420bf6fc0bf9a6963dbd72edff8f5bdb3a99ce45 Mon Sep 17 00:00:00 2001
From: Nikolche Kolev <nikolev@microsoft.com>
Date: Mon, 30 Oct 2017 13:31:52 -0700
Subject: [PATCH] Remove NuGet patches (#264)

* Remove patches and change calls to build proj

* Update .gitmodules

* remove all nuget-client patches

* pass configuration parameters

* Update .gitmodules

* update submodule
---
 patches/nuget-client/.gitattributes                |   4 -
 ...9ce93081016167dfd8e37e4e897bedf47d65-DOS-.patch |  75 ------
 ...9ce93081016167dfd8e37e4e897bedf47d65-Unix.patch |  82 -------
 patches/nuget-client/0003-Fix-Nuspec.patch         |  31 ---
 ...-netstandard-and-netcoreapp-assets-on-NIX.patch | 193 ----------------
 ...ln-which-only-includes-projects-CLI-needs.patch | 256 ---------------------
 patches/nuget-client/0006-Fix-signing.patch        |  36 ---
 .../nuget-client/0007-Bump-MSBuild-Reference.patch |  25 --
 repos/nuget-client.proj                            |   7 +-
 src/nuget-client                                   |   2 +-
 10 files changed, 6 insertions(+), 705 deletions(-)
 delete mode 100644 patches/nuget-client/.gitattributes
 delete mode 100644 patches/nuget-client/0001-Revert-38949ce93081016167dfd8e37e4e897bedf47d65-DOS-.patch
 delete mode 100644 patches/nuget-client/0002-Revert-38949ce93081016167dfd8e37e4e897bedf47d65-Unix.patch
 delete mode 100644 patches/nuget-client/0003-Fix-Nuspec.patch
 delete mode 100644 patches/nuget-client/0004-Only-build-netstandard-and-netcoreapp-assets-on-NIX.patch
 delete mode 100644 patches/nuget-client/0005-Add-CLI.sln-which-only-includes-projects-CLI-needs.patch
 delete mode 100644 patches/nuget-client/0006-Fix-signing.patch
 delete mode 100644 patches/nuget-client/0007-Bump-MSBuild-Reference.patch

diff --git a/patches/nuget-client/.gitattributes b/patches/nuget-client/.gitattributes
deleted file mode 100644
index 423431d..0000000
diff --git a/patches/nuget-client/0001-Revert-38949ce93081016167dfd8e37e4e897bedf47d65-DOS-.patch b/patches/nuget-client/0001-Revert-38949ce93081016167dfd8e37e4e897bedf47d65-DOS-.patch
deleted file mode 100644
index e594bd4..0000000
diff --git a/patches/nuget-client/0002-Revert-38949ce93081016167dfd8e37e4e897bedf47d65-Unix.patch b/patches/nuget-client/0002-Revert-38949ce93081016167dfd8e37e4e897bedf47d65-Unix.patch
deleted file mode 100644
index fa62398..0000000
diff --git a/patches/nuget-client/0003-Fix-Nuspec.patch b/patches/nuget-client/0003-Fix-Nuspec.patch
deleted file mode 100644
index 22374b9..0000000
diff --git a/patches/nuget-client/0004-Only-build-netstandard-and-netcoreapp-assets-on-NIX.patch b/patches/nuget-client/0004-Only-build-netstandard-and-netcoreapp-assets-on-NIX.patch
deleted file mode 100644
index 9fd5c03..0000000
diff --git a/patches/nuget-client/0005-Add-CLI.sln-which-only-includes-projects-CLI-needs.patch b/patches/nuget-client/0005-Add-CLI.sln-which-only-includes-projects-CLI-needs.patch
deleted file mode 100644
index 4e83046..0000000
diff --git a/patches/nuget-client/0006-Fix-signing.patch b/patches/nuget-client/0006-Fix-signing.patch
deleted file mode 100644
index 13a9770..0000000
diff --git a/patches/nuget-client/0007-Bump-MSBuild-Reference.patch b/patches/nuget-client/0007-Bump-MSBuild-Reference.patch
deleted file mode 100644
index 564cbd4..0000000
diff --git a/repos/nuget-client.proj b/repos/nuget-client.proj
index aa59511..d0212c3 100644
--- a/repos/nuget-client.proj
+++ b/repos/nuget-client.proj
@@ -21,10 +21,13 @@
   <Target Name="RepoBuild"
           DependsOnTargets="InitSubmodules">
 
-    <Exec Command="$(DotnetToolCommand) restore $(ProjectDirectory)/CLI.sln /p:BuildRTM=false /p:BuildNumber=$(NuGetClientBuildNumber) $(RedirectRepoOutputToLog)"
+    <Exec Command="$(DotnetToolCommand) msbuild $(ProjectDirectory)/build/build.proj /t:RestoreXPLAT /p:VisualStudioVerion=15.0 /p:Configuration=$(Configuration) /p:BuildRTM=false /p:BuildNumber=$(NuGetClientBuildNumber) $(RedirectRepoOutputToLog)"
           EnvironmentVariables="@(EnvironmentVariables)" />
 
-    <Exec Command="$(DotnetToolCommand) pack $(ProjectDirectory)/CLI.sln /p:VisualStudioVerion=15.0 /p:Configuration=$(Configuration) /p:BuildRTM=false /p:BuildNumber=$(NuGetClientBuildNumber) /p:PackageOutputPath=$(PackagesOutput) /p:NoPackageAnalysis=true /flp:v=detailed $(RedirectRepoOutputToLog)"
+    <Exec Command="$(DotnetToolCommand) msbuild $(ProjectDirectory)/build/build.proj /t:BuildXPLAT /p:VisualStudioVerion=15.0 /p:Configuration=$(Configuration) /p:BuildRTM=false /p:BuildNumber=$(NuGetClientBuildNumber) $(RedirectRepoOutputToLog)"
+          EnvironmentVariables="@(EnvironmentVariables)" />
+
+    <Exec Command="$(DotnetToolCommand) msbuild $(ProjectDirectory)/build/build.proj /t:PackXPLAT /p:VisualStudioVerion=15.0 /p:Configuration=$(Configuration) /p:BuildRTM=false /p:BuildNumber=$(NuGetClientBuildNumber) /p:PackageOutputPath=$(PackagesOutput) /p:NoPackageAnalysis=true /flp:v=detailed $(RedirectRepoOutputToLog)"
           EnvironmentVariables="@(EnvironmentVariables)" />
   </Target>
 
diff --git a/src/nuget-client b/src/nuget-client
index 884b3d8..93d8523 160000
--- a/src/nuget-client
+++ b/src/nuget-client
@@ -1 +1 @@
-Subproject commit 884b3d8f10656137babc06db85dac63d10757bbb
+Subproject commit 93d85237e4c21ade2376b2b49cc8822852c1b3c7
-- 
2.7.4

